BDArmory/Control/BDModulePilotAI.cs

```diff
@@ -4011,6 +4018,34 @@ void ResetGradient()
                         fixedFields.Add(field.name, field.guiName);
                         continue;
                     }
+                    // Exclude relevant damping fields when disabled
+                    if (AI.dynamicSteerDamping)
+                    {
+                        if ((!AI.CustomDynamicAxisFields || (AI.CustomDynamicAxisFields && AI.dynamicDampingPitch && AI.dynamicDampingYaw && AI.dynamicDampingRoll)) &&
+                        field.name == "steerDamping")
+                        {
+                            fixedFields.Add(field.name, field.guiName);
+                            continue;
+                        }
+                        else if (AI.CustomDynamicAxisFields)
+                        {
+                            if ((!AI.dynamicDampingPitch) && ((field.name == "DynamicDampingPitchMin") || (field.name == "DynamicDampingPitchMax") || (field.name == "DynamicDampingPitchFactor")))
```

Comment:
The latter part of this check can be done with `field.name.StartsWith("DynamicDampingPitch")` instead of 3 separate comparisons. Similarly for the others below.